app/src/main/java/com/example/mystore/ui/home/HomeViewModel.kt

```diff
@@ -90,8 +92,9 @@ class HomeViewModel @Inject constructor(
             isLoading.value = true
 
             try {
-                allProducts.update { emptyList() }
-                allProducts.value = apiRepository.getProducts()
+                allProducts.value.apply {
```

Comment:
Same comment here - the `apply` here is unnecessary and as it is it does nothing except exposing a `this` which you do not use.